Classes/Security/CheckPermissions.php

```diff
@@ -256,6 +266,16 @@ public function getFolderRootLine(FolderInterface $folder): array
         return array_reverse($rootLine);
     }
 
+    /**
+     * Get all folders in root line of given folder
+     *
+     * @return Folder[]
+     */
+    public function getFolderRootLine(FolderInterface $folder)
+    {
+        return array_reverse($this->getReverseFolderRootLine($folder));
```

Comment:
This would mean a double `array_reverse`. It's better to do it the other way around. That way you'll only need one `array_reverse`. So move the code from `getReverseFolderRootline` to `getFolderRootline`, remove the `array_reverse` on the return and change `getReverseFolderRootline` to `return array_reverse($this->getFolderRootline($folder));`

Reply:
No, there's only this one array_reverse() in all the code. The method $this->getReverseFolderRootLine() builts the reverse array native.

Reply:
Oops, in my patched version ist was removed. Forgot it here.
Thank you for your attention :-)